shared/src/commonMain/kotlin/org/hyperskill/app/study_plan/widget/presentation/StudyPlanWidgetReducer.kt

```diff
@@ -188,10 +188,27 @@ class StudyPlanWidgetReducer : StateReducer<State, Message, Action> {
                 val activitiesIdsToRemove = activitiesIds - message.activities.map { it.id }.toSet()
                 activitiesIdsToRemove.forEach { remove(it) }
             },
-            studyPlanSections = state.studyPlanSections.update(message.sectionId) { sectionInfo ->
-                sectionInfo.copy(contentStatus = StudyPlanWidgetFeature.ContentStatus.LOADED)
+            // ALTAPPS-786: We should hide sections without available activities to avoid blocking study plan
+            studyPlanSections = if (message.activities.isNotEmpty()) {
+                state.studyPlanSections.update(message.sectionId) { sectionInfo ->
+                    sectionInfo.copy(contentStatus = StudyPlanWidgetFeature.ContentStatus.LOADED)
+                }
+            } else {
+                state.studyPlanSections.toMutableMap().apply {
+                    remove(message.sectionId)
+                }
             }
-        ) to emptySet()
+        )
+
+        // ALTAPPS-786: We should expand next section if current section doesn't have available activities
+        return if (message.sectionId == state.getCurrentSection()?.id && message.activities.isEmpty()) {
+            state.studyPlanSections.keys.drop(1).firstOrNull()?.let { nextSectionId ->
```

Comment:
Just a few lines above you removed the study plan section if it has empty learning activities (key and its corresponding value):
```kotlin
state.studyPlanSections.toMutableMap().apply {
    remove(message.sectionId)
}
```

So, you can use `nextState` to get the current section (without use of `drop(1)`):
```kotlin
nextState.studyPlanSections.keys.firstOrNull()?.let { nextSectionId ->
    changeSectionExpanse(nextState, nextSectionId, shouldLogAnalyticEvent = false)
} ?: (nextState to emptySet())
```